src/validator-info.js

```diff
@@ -0,0 +1,100 @@
+// @flow
+
+import {struct} from 'superstruct';
+
+import * as Layout from './layout';
+import * as shortvec from './util/shortvec-encoding';
+import {PublicKey} from './publickey';
+
+const VALIDATOR_INFO_KEY = new PublicKey(
+  'Va1idator1nfo111111111111111111111111111111',
+);
+
+/**
+ * @private
+ */
+type ConfigKey = {|
+  publicKey: PublicKey,
+  isSigner: boolean,
+|};
+
+/**
+ * Info used to identity validators.
+ *
+ * @typedef {Object} Info
+ * @property {string} name validator name
+ * @property {?string} website optional, validator website
+ * @property {?string} details optional, extra information the validator chose to share
+ * @property {?string} keybaseId optional, used to identify validators on keybase.io
+ */
+export type Info = {|
+  name: string,
+  website?: string,
+  details?: string,
+  keybaseId?: string,
+|};
+
+const InfoString = struct({
+  name: 'string',
+  website: 'string?',
+  details: 'string?',
+  keybaseId: 'string?',
+});
+
+/**
+ * ValidatorInfo class
+ */
+export class ValidatorInfo {
+  /**
+   * validator public key
+   */
+  key: PublicKey;
+  /**
+   * validator information
+   */
+  info: Info;
+
+  /**
+   * Construct a valid ValidatorInfo
+   *
+   * @param key validator public key
+   * @param info validator information
+   */
+  constructor(key: PublicKey, info: Info) {
+    this.key = key;
+    this.info = info;
+  }
+
+  /**
+   * Deserialize ValidatorInfo from the config account data.
+   *
+   * @param buffer config account data
+   * @return null if info was not found
+   */
+  static fromConfigData(buffer: Buffer): ?ValidatorInfo {
+    const PUBKEY_LENGTH = 32;
+
+    let byteArray = [...buffer];
+    const configKeyCount = shortvec.decodeLength(byteArray);
+    if (configKeyCount < 2) return null;
```

Comment:
I'd bail if `configKeyCount !== 2`, since it looks like we'll go off the rails below if there are more than 2 keys (`byteArray` won't be in the right place)

Reply:
oops, good catch